sklearn/manifold/_t_sne.py

```diff
@@ -666,6 +678,16 @@ def _fit(self, X, skip_num_points=0):
             raise ValueError("'method' must be 'barnes_hut' or 'exact'")
         if self.angle < 0.0 or self.angle > 1.0:
             raise ValueError("'angle' must be between 0.0 - 1.0")
+        if self.square_distance not in [True, 'legacy']:
+            raise ValueError("'square_distance' must be True or 'legacy'.")
+        if self.metric != "euclidean" and self.square_distance is not True:
+            warnings.warn(("'square_distance' has been introduced in 0.24. "
+                           "'legacy' is provided as a setting for backward "
+                           "compatibility purposes. However, 'legacy' will "
+                           "be removed in 0.26, at which point all distance "
+                           "metrics will be squared by default. Set "
+                           "'square_distance' to True to silence this "
+                           "warning."), FutureWarning)
```

Comment:
I would remove the last sentence because setting to True changes the behavior in addition to silence the warning.